src/session.rs

```diff
@@ -97,19 +126,35 @@ impl Content {
                     .collect(),
             );
         }
+        if let Cookies::Map(cookies) = self.cookies {
+            self.cookies = Cookies::List(
+                cookies
+                    .into_iter()
+                    .map(|(name, legacy_cookie)| Cookie {
+                        name,
+                        value: legacy_cookie.value,
+                        expires: legacy_cookie.expires,
+                        path: legacy_cookie.path,
+                        secure: legacy_cookie.secure,
+                        domain: None,
+                    })
+                    .collect(),
+            )
+        }
 
         self
     }
 }
 
-pub struct Session {
+pub struct Session<'a> {
+    url: &'a Url,
```

Comment:
I think it'd be easier to just clone this one, we're not creating lots of `Session`s.